src/runtime/audience-action-local-flow.ts

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright 2018 The Subscribe with Google Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * An audience action local flow will show a dialog prompt to a reader, asking them
+ * to complete an action for potentially free, additional metered entitlements.
+ */
+
+import {AudienceActionFlow} from './audience-action-flow';
+import {AutoPromptType} from '../api/basic-subscriptions'; // @ts-ignore
+import {ClientConfigManager} from './client-config-manager';
+import {Deps} from './deps';
+import {SWG_I18N_STRINGS} from '../i18n/swg-strings';
+import {Toast} from '../ui/toast';
+import {createElement} from '../utils/dom';
+import {feUrl} from './services';
+import {msg} from '../utils/i18n';
+import {setImportantStyles} from '../utils/style';
+
+// Helper for syntax highlighting.
+const html = String.raw;
+const css = String.raw;
+
+const INVALID_CSS = css`
```

Comment:
Could we good to add a comment saying it's for development / test only.

Reply:
Done.

Reply:
Will this be removed as implementation progresses? If so can you add a TODO to clean up?